core/src/smartcontracts/isi/domain.rs

```diff
@@ -105,7 +109,12 @@ pub mod isi {
                 .world
                 .domain_mut(&account_id.domain)?
                 .remove_account(&account_id)
-                .is_none()
+                == false
+                || state_transaction
+                    .world
+                    .accounts
+                    .remove(account_id.clone())
+                    .is_none()
```

Comment:
This shouldn't fail if the check to the left of `||` has succeeded, otherwise iroha's storage is in some inconsistent state. Should this be an assert?

Reply:
When writing an application I would assert. But here I think we want to do our best without crashing. Perhaps I can minimize the potential for this inconsistent state.